src/pywbem/twisted_client.py

```diff
@@ -300,6 +302,109 @@ def parseResponse(self, xml):
         return [pywbem.tupleparse.parse_instance(x) for x in tt]
 
 
+class OpenEnumerateInstances(WBEMClientFactory):
+    """Factory to produce EnumerateInstances WBEM clients."""
+
+    def __init__(self, creds, classname, namespace='root/cimv2', **kwargs):
+        self.classname = classname
+        self.namespace = namespace
+        self.context = None
+
+        if not kwargs.get('MaxObjectCount'):
+            kwargs['MaxObjectCount'] = DEFAULT_ITER_MAXOBJECTCOUNT
+
+        payload = self.imethodcallPayload(
+            'OpenEnumerateInstances',
+            namespace,
+            ClassName=CIMClassName(classname),
+            **kwargs)
+
+        WBEMClientFactory.__init__(
+            self,
+            creds,
+            operation='MethodCall',
+            method='OpenEnumerateInstances',
+            object=namespace,
+            payload=payload)
+
+    def __repr__(self):
+        return '<%s(/%s:%s) at 0x%x>' % \
+               (self.__class__, self.namespace, self.classname, id(self))
+
+    def parseResponse(self, xml):
+        res = []
+        part_results = {}
+
+        for paramvalue in xml.findall('.//PARAMVALUE'):
+            str_paramvalue = tostring(paramvalue)
+            tuple_paramvalue = pywbem.tupletree.xml_to_tupletree(str_paramvalue)
+            part_results.update(pywbem.tupleparse.parse_paramvalue_to_dict(tuple_paramvalue))
+
+        for x in xml.findall('.//VALUE.INSTANCEWITHPATH'):
+            s = tostring(x)
+            tt = pywbem.tupletree.xml_to_tupletree(s)
+            part_res = pywbem.tupleparse.parse_value_instancewithpath_to_dict(tt)
+            res.append(part_res['VALUE.INSTANCEWITHPATH'])
+
+        part_results.update({'IRETURNVALUE': res})
+        return self._get_rslt_params(part_results)
+
+
+    @staticmethod
+    def _get_rslt_params(result):
+        """Common processing for pull results to separate
+           end-of-sequence, enum-context, and entities in IRETURNVALUE.
+           Returns tuple of entities in IRETURNVALUE, end_of_sequence,
+           and enumeration_context)
+        """
+        end_of_sequence = False
+        enumeration_context = None
+
+        sequence = result.get('EndOfSequence')
+        if sequence and isinstance(sequence, six.string_types) and \
+                        sequence.lower() in ['true', 'false']:  # noqa: E125
+            end_of_sequence = True if sequence.lower() == 'true' else False
```

Comment:
Just to simplify you can use: `sequence.lower() in ("true")`

Reply:
hmm, it is copied from the new pywbem library, but yes it is better to replace it with `sequence.lower() == 'true'`